src/Universum/Lifted/Concurrent.hs

```diff
@@ -108,3 +115,26 @@ newTVarIO = liftIO . STM.newTVarIO
 readTVarIO :: MonadIO m => TVar a -> m a
 readTVarIO = liftIO . STM.readTVarIO
 {-# INLINE readTVarIO #-}
+
+----------------------------------------------------------------------------
+-- Common helpers
+----------------------------------------------------------------------------
+
+-- | Like 'modifyMVar', but modification is specified as a 'State' computation.
+--
+-- This method is strict in produced @s@ value.
+updateMVarS' :: MonadIO m => MVar s -> StateT s IO a -> m a
+updateMVarS' var (StateT f) =
+  liftIO . CCM.modifyMVar var $ \s -> do
+    (!s', a) <- f s
+    return (a, s')
+{-# INLINE updateMVarS' #-}
+
+-- | Like 'modifyTVar'', but modification is specified as a 'State' monad.
+updateTVar' :: TVar s -> StateT s STM a -> STM a
+updateTVar' var (StateT f) = do
```

Comment:
Is naming intentionally inconsistent between `updateTVar'` and `updateMVarS'`?

Reply:
Meh no, forgot to remove `S` suffix.

Reply:
pinging @Martoon-00  ^^

Reply:
Forgot to push the changes :sweat_smile: